lib/matplotlib/cbook/__init__.py

```diff
@@ -1645,19 +1662,20 @@ def pts_to_midstep(x, *args):
 
 def index_of(y):
     """
-    A helper function to get the index of an input to plot
-    against if x values are not explicitly given.
+    A helper function to create reasonable x values for the given *y*.
+
+    This is used for plotting (x, y) if x values are not explicitly given.
 
-    Tries to get `y.index` (works if this is a pd.Series), if that
-    fails, return np.arange(y.shape[0]).
+    First try ``y.index`` (assuming *y* is a `pandas.Series`), if that
+    fails, use integer numbers 0...n counting the elements of *y*.
 
     This will be extended in the future to deal with more types of
     labeled data.
 
     Parameters
     ----------
     y : scalar or array-like
-        The proposed y-value
+        The proposed y-value.
```

Comment:
I would remove this description (which has... limited semantic information, to say the least ("y is the proposed y value", duh))